sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/literals.scala

```diff
@@ -60,6 +76,45 @@ object Literal {
   }
 
   /**
+   * Returns the Spark SQL DataType for a given class object. Since this type needs to be resolved
+   * in runtime, we use match-case idioms for class objects here. However, there are similar
+   * functions in other files (e.g., HiveInspectors), so these functions need to merged into one.
+   */
+  private[this] def componentTypeToDataType(clz: Class[_]): DataType = clz match {
+    // primitive types
+    case c: Class[_] if c == JavaShort.TYPE => ShortType
```

Comment:
cool, I saw you updated the match. However you can remove the instance check everywhere, including further down. Basically a `case c: Class[_] ` is "equivalent" to an `c.isInstanceOf[Class[_]]`, however that is redundant since your parameter `clz` already specifies the type to be `Class[_]`